### PR TITLE
Added unit tests for MicroSwitchController class

### LINKED ISSUE #302: Write Unit Tests for MicroSwitchHelper Class
**Description:**
  The MicroSwitchHelper class manages the state of a micro switch using Pi4J's DigitalInput and event listeners. Unit tests are needed to verify the correctness of its functionality, particularly for event listener management, state updates, and logging behavior.

**Tasks:**

1. Test Initialization Logic:

    Verify that the isPressed field is correctly initialized based on the initial state of the DigitalInput object passed to the constructor.

2. Test Event Listener Management:

    Confirm that the listener is properly added using the addEventListener method.
    Ensure that the correct listener is triggered when the micro switch's state changes (i.e., when the switch is pressed or released).
    Validate that the listener is correctly removed using the removeEventListener method and that the microSwitchInputListener is set to null afterward.

3. Test State Update:

    Verify that the isPressed field is updated correctly when the micro switch state changes.

4. Test Logging:

    Ensure that the initialize method logs the expected message ("Initializing Micro Switch").

5. Mocking Dependencies:

    Use a mocking framework (e.g., Mockito) to mock the DigitalInput and DigitalStateChangeListener dependencies.

6. Edge Cases:

    Test the behavior when no event listener is attached, but the state of the micro switch changes.
    Verify that the removeEventListener method works as expected when no listener has been added.

**Acceptance Criteria**

- [x] All test cases outlined in the tasks above are implemented.
- [x] Code coverage for the MicroSwitchHelper class is at least 70%.
- [x] Tests are written using a Java JUnit testing framework
- [x] Mocking is implemented using a framework such as Mockito or a similar tool.
- [x] All tests pass successfully, and edge cases are handled gracefully.

### PR DESCRIPTION
**Pull request by avaenk**
Fixes #302 

**Description**: 
Adds unit tests for the MicroSwitchHelper using JUnit 5 and Mockito, checking its initialization, state updates and event listener management. Also introduces logger injection to confirm the correct log message ('Initializing Micro Switch') is being produced. 

What was **changed**: 
In MicroSwitchHelper the logger was converted from static final to static with a new setLog(Logger newLog) method, this enables the use of a custom logger for testing. 
Unit tests were implemented to verify initialization logic, state updates, and event listener management, and tests to confirm the expected log message. 

How it was **changed**:
The logger in MicroSwitchHelper was changed from a static final to an instance level with the new setLog method. Unit tests were implemented using JUnit 5 and Mockito in order to simulate different kinds of digital input and to verify specific fields are correct upon initialization. Verified the initial listener in the initialize methods updates isPressed state correctly when the input state is changed. New tests to confirm that the add and remove event listener methods work correctly. New logging tests to check that the expected log messa

Screenshot 2025-02-02 at 1 55 56 PM

### CODE REVIEW

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/inputdevices/MicroSwitchHelper.java [hunk 1]
```diff
@@ -64,4 +64,9 @@ public void removeEventListener()
             microSwitchInputListener = null;
         }
     }
+
+    public void setLog(Logger log) {
```
Comment (Ed0827) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/inputdevices/MicroSwitchHelper.java:68]: Great job on this! This will help when troubleshooting your code by setting up a logger.

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/inputdevices/MicroSwitchHelper.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
  */
 public class MicroSwitchHelper {
 
-    private static final Logger log = LoggerFactory.getLogger(MicroSwitchHelper.class);
```
Comment (Ed0827) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/inputdevices/MicroSwitchHelper.java:13]: This provides flexibility to modify logger messages.

**Review by Ed0827 — COMMENTED** [review]
The functionalities provided in the helper are well organized and utilized effectively. Great job!

**Review by yrlmanoharreddy — CHANGES_REQUESTED** [review]
Close this pull request.
